src/settings/appsettings.cpp

```diff
@@ -664,6 +664,51 @@ QOnlineTts::Emotion AppSettings::defaultEmotion(QOnlineTranslator::Engine engine
     }
 }
 
+QMap<QOnlineTranslator::Language, QLocale::Country> AppSettings::regions(QOnlineTranslator::Engine engine) const
+{
+    switch (engine) {
+    case QOnlineTranslator::Google: {
+        QMap<QOnlineTranslator::Language, QLocale::Country> regions;
+        for (const QOnlineTranslator::Language lang : QOnlineTts::validRegions().keys())
+            regions[lang] = m_settings->value(QString("TTS/GoogleRegion/%1").arg(QOnlineTranslator::languageName(lang)), QLocale::AnyCountry).value<QLocale::Country>();
+        return regions;
+    }
+    case QOnlineTranslator::Bing:
+    case QOnlineTranslator::Yandex:
+    case QOnlineTranslator::LibreTranslate:
+    case QOnlineTranslator::Lingva:
+        return {};
+    default:
+        Q_UNREACHABLE();
+    }
+}
+
+void AppSettings::setRegions(QOnlineTranslator::Engine engine, const QMap<QOnlineTranslator::Language, QLocale::Country> &regions)
+{
+    switch (engine) {
+    case QOnlineTranslator::Google:
+        for (auto it = regions.cbegin(); it != regions.cend(); ++it)
+            m_settings->setValue(QString("TTS/GoogleRegion/%1").arg(QOnlineTranslator::languageName(it.key())), it.value());
+        return;
```

Comment:
A thing to consider. It currently displayed like this:
```dosini
[TTS]
GoogleRegion\Bengali=0
GoogleRegion\Chinese%20%28Simplified%29=0
GoogleRegion\English=224
GoogleRegion\French=74
GoogleRegion\German=0
GoogleRegion\Portuguese=0
GoogleRegion\Spanish=0
GoogleRegion\Tamil=0
YandexEmotion=0
YandexVoice=0
```

But what if we try to store it as `QMap<QString, QVariant>` (where QVariant can be any type)) which is supported?
I.e. instead of using `QOnlineTranslator::languageName(lang)` to create a path use it to create a string map and feed it to QSettings via `setValue`. And the same for getting. I.e. perform a conversion.

Reply:
Now the config file looks like this, which is not human-readable at all.
```ini
[TTS]
GoogleRegions=@Variant(\0\0\0\b\0\0\0\b\0\0\0\n\0T\0\x61\0m\0i\0l\0\0\0\x2\0\0\0\0\0\0\0\xe\0S\0p\0\x61\0n\0i\0s\0h\0\0\0\x2\0\0\0\0\0\0\0\x14\0P\0o\0r\0t\0u\0g\0u\0\x65\0s\0\x65\0\0\0\x2\0\0\0\0\0\0\0\f\0G\0\x65\0r\0m\0\x61\0n\0\0\0\x2\0\0\0\0\0\0\0\f\0\x46\0r\0\x65\0n\0\x63\0h\0\0\0\x2\0\0\0\0\0\0\0\xe\0\x45\0n\0g\0l\0i\0s\0h\0\0\0\x2\0\0\0\xe0\0\0\0(\0\x43\0h\0i\0n\0\x65\0s\0\x65\0 \0(\0S\0i\0m\0p\0l\0i\0\x66\0i\0\x65\0\x64\0)\0\0\0\x2\0\0\0\0\0\0\0\xe\0\x42\0\x65\0n\0g\0\x61\0l\0i\0\0\0\x2\0\0\0\0)
YandexEmotion=0
YandexVoice=0
```

But anyway, configs are not meant for us humans :) and my original version still requires a Qt manual at hand.

Reply:
Hm, right, not much better :smile: 
Yes, let's keep it.